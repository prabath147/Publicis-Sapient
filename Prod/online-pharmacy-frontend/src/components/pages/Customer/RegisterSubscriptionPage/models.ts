
export interface RegisterSubscription {
    subscriptionId: number,
    subscriptionName: string,
    description: string,
    startDate: Date,
    endDate: Date,
    benefits: {
        delivery_discount: number,
        one_day_delivery: boolean, //enable-disable
    },
    subscriptionCost: number,
    period: number,
    status: string //active-expired
}

export interface Unsubscribe {
    user_id: number,
    subscription_id: string
}

export interface Subscriber {
    userId: number,
    userSubsSet: UserSubs[]
}

export interface UserSubs {
    userSubId: number,
    userId: number,
    subscriptions: RegisterSubscription,
    lastPaidDate: Date,
    subEndDate: Date,
    status: string
}