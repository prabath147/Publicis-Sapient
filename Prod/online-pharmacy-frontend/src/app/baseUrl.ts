export const baseURL = "https://frk5iphcfk.execute-api.us-east-1.amazonaws.com/";
    // "http://localhost:8081";
    
