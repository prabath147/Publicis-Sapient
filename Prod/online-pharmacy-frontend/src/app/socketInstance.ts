
export const socketURL = "https://c8zpvs7fv4.execute-api.us-east-1.amazonaws.com/notify-ws"

// const socket = new SockJS(socketURL);
// export const stompClient = Stomp.over(socket);