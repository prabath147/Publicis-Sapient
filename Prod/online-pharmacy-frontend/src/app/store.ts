import { Action, configureStore, ThunkAction } from "@reduxjs/toolkit";
import StoreReducer from "../components/pages/Admin/storeList/StoreSlice";
import SubscriptionReducer from "../components/pages/AdminSubscriptionPack/SubscriptionvSlice";
import UserReducer from "../components/pages/Auth/Login/UserSlice";
import CartReducer from "../components/pages/Cart/CartSlice";
import ManagerReducer from "../components/pages/Manager/ManagerSlice";
import NotificationReducer from "../components/pages/Open/notificationList/NotificationSlice";
// import OrderSlice from "../components/pages/orderDetail/OrderSlice";
// import OrderListSlice from "../components/pages/orderList/OrderListSlice";

export const store = configureStore({
  reducer: {
    storeL: StoreReducer,
    user: UserReducer,
    notification: NotificationReducer,
    viewsubscription: SubscriptionReducer,
    // addsubscription: AddSubcriptionRe,
    cart: CartReducer,
    // order: OrderSlice,
    // orderL: OrderListSlice,
    manager: ManagerReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
