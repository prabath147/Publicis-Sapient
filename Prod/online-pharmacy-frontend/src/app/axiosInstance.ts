import axios, { AxiosRequestConfig } from "axios";
import LocalUserService from "../services/LocalUserService";
import TokenService from "../services/TokenService";
import { baseURL } from "./baseUrl";

const axiosInstance = axios.create({
  baseURL: baseURL,
  timeout: 5000,
  headers: {
    "Content-Type": "application/json",
    accept: "application/json",
  },
});

axiosInstance.interceptors.request.use(
  (config: AxiosRequestConfig) => {
    const token = TokenService.getLocalAccessToken();
    if (token && config.headers !== undefined) {
      config.headers["Authorization"] = "Bearer " + token; // for Spring Boot back-end
      // config.headers["x-access-token"] = token; // for Node.js Express back-end
      config.headers["ngrok-skip-browser-warning"] = "";
    }
    return config;
  },
  (error) => {
    return Promise.reject(error);
  }
);

axiosInstance.interceptors.response.use(
  (res) => {
    return res;
  },
  async (err) => {
    const originalConfig = err.config;
    if (originalConfig.url !== "api/auth/signin" && err.response) {
      // Access Token was expired
      if (err.response.status === 401 && !originalConfig._retry) {
        originalConfig._retry = true;
        try {
          const rs = await axiosInstance.post("api/auth/refresh-token", {
            refreshToken: TokenService.getLocalRefreshToken(),
          });
          const { accessToken } = rs.data;
          TokenService.updateLocalAccessToken(accessToken);
          return axiosInstance(originalConfig);
        } catch (_error) {
          TokenService.removeLocalAccessToken()
          TokenService.removeLocalRefreshToken()
          LocalUserService.removeLocalUser()
          // window.location.reload();
          return Promise.reject(_error);
        }
      }
    }
    return Promise.reject(err);
  }
);
export default axiosInstance;
