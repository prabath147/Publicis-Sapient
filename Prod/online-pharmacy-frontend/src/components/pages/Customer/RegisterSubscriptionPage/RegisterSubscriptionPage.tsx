import {
  Badge,
  Button,
  Card,
  Center,
  Container,
  Grid,
  Group,
  LoadingOverlay,
  Pagination,
  Text,
  Title
} from "@mantine/core";
import { showNotification } from "@mantine/notifications";
import { useEffect, useState } from "react";
import { useAppSelector } from "../../../../app/hooks";
import { getUserData } from "../../Auth/Login/UserSlice";
import { RegisterSubscription } from "./models";
import {
  getSubscription,
  subscribeAPI,
  usersubscriptionAPI
} from "./SubscriptionAPI";

export default function RegisterSubscriptionPage() {
  // const [selectedId, changeSelection] = useState(0);
  const [data, setData] = useState<RegisterSubscription[]>([]);
  const [loading, setLoading] = useState(true);
  const user = useAppSelector(getUserData);
  const [subscriptions, setSubscriptions] = useState([]);

  const [pageCount, setPageCount] = useState(1);
  const [activePage, setPage] = useState(1);

  useEffect(() => {
    getSubscription(activePage - 1, 9)
      .then((response) => {
        // console.log("output = " + response.data.data);
        setData(response.data.data);
        setPageCount(response.data.totalPages);
      })
      .catch((error) => {
        console.log("Error" + error.message);
      });

    usersubscriptionAPI(user.id)
      .then((response) => {
        // console.log(response.data.userSubsSet);
        let userSubscriptions = [];
        userSubscriptions = response.data.userSubsSet.map((item) => {
          return item.subscriptions.subscriptionId;
        });
        // console.log(userSubscriptions);
        setSubscriptions(userSubscriptions);
      })
      .catch((error) => {
        console.log("Error" + error.message);
      });
    setLoading(false);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [activePage]);

  const handleClick = (id) => {
    // changeSelection(id);
    subscribeAPI(user.id, id)
      .then((response) => {
        showNotification({
          message: "Subscribed!",
          color: "green",
          autoClose: 500,
        });
        getSubscription(activePage - 1, 9)
          .then((response) => {
            console.log("output = " + response.data.data);
            setData(response.data.data);
          })
          .catch((error) => {
            console.log("Error" + error.message);
          });
        usersubscriptionAPI(user.id)
          .then((response) => {
            console.log(response.data.userSubsSet);
            let userSubscriptions = [];
            userSubscriptions = response.data.userSubsSet.map((item) => {
              return item.subscriptions.subscriptionId;
            });
            console.log(userSubscriptions);
            setSubscriptions(userSubscriptions);
          })
          .catch((error) => {
            console.log("Error" + error.message);
          });
      })
      .catch((error) => {
        console.log("Error" + error.message);
        showNotification({
          message: "Something went wrong, please try again!",
          color: "red",
        });
      });
  };

  const status = (item: RegisterSubscription) => {
    const arr: number[] = subscriptions;
    if (arr.includes(item.subscriptionId)) {
      return (
        <Button
          size="xs"
          variant="gradient"
          gradient={{ from: "teal", to: "lime" }}
          disabled
        >
          Subscribe
        </Button>
      );
    } else {
      return (
        <Button
          size="xs"
          variant="gradient"
          onClick={() => handleClick(item.subscriptionId)}
          gradient={{ from: "teal", to: "lime" }}
        >
          Subscribe
        </Button>
      );
    }
  };

  return (
    <>
      {" "}
      {loading ? (
        <LoadingOverlay
          data-testid="Loading"
          loaderProps={{ size: "md", color: "blue", variant: "oval" }}
          overlayOpacity={0.3}
          overlayColor="#c5c5c5"
          visible={loading}
        />
      ) : (
        <Container>
          <Title align="center" order={2}>
            Available Subscriptions
          </Title>
          <br />
          <Center>
            <Grid columns={90} sx={{ width: "100%" }}>
              {data.map((item) => {
                return (
                  <Grid.Col sm={60} md={30}>
                    <Card
                      shadow="sm"
                      p="md"
                      style={{
                        width: "100%",
                        height: 300,
                        marginTop: "10px",
                        marginBottom: "10px",
                      }}
                      radius="md"
                      withBorder
                    >
                      <Text
                        style={{ fontSize: 22, fontWeight: 700, marginTop: 10 }}
                      >
                        {item.subscriptionName}
                        <br />
                        {"â‚¹" + item.subscriptionCost}
                      </Text>

                      <Group sx={{ width: "100%" }} position="apart">
                        <Text
                          size="md"
                          weight={700}
                          style={{ marginBottom: 10 }}
                        >
                          <Badge>{item.startDate + "-" + item.endDate}</Badge>
                        </Text>
                      </Group>

                      <div style={{ height: "70px" }}>
                        <Group style={{ marginBottom: 10 }} spacing="xs">
                          {item.benefits.one_day_delivery === true ? (
                            <Badge
                              variant="gradient"
                              gradient={{ from: "#bd0000", to: "red" }}
                            >
                              1 DAY DELIVERY
                            </Badge>
                          ) : (
                            ""
                          )}
                          {item.benefits.delivery_discount !== 0 ? (
                            <Badge
                              variant="gradient"
                              gradient={{ from: "#bd0000", to: "red" }}
                            >
                              {item.benefits.delivery_discount +
                                "% OFF ON DELIVERY"}
                            </Badge>
                          ) : (
                            ""
                          )}
                        </Group>
                      </div>
                      <Text lineClamp={2}>{item.description}</Text>
                      <Center mt="xs">{status(item)}</Center>
                    </Card>
                  </Grid.Col>
                );
              })}
            </Grid>
          </Center>
          <br />
          {/* <Center>
      <Button size="sm" onClick={() => console.log(selectedId)}>
        Submit
      </Button>
    </Center> */}
        </Container>
      )}
      <Pagination
        mb="lg"
        data-testid="paginationT"
        position="center"
        page={activePage}
        onChange={setPage}
        total={pageCount}
      />
    </>
  );
}
