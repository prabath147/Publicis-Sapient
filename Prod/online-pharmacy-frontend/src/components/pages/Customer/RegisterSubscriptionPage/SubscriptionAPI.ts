import axiosInstance from "../../../../app/axiosInstance";


export const getSubscription = (pageNo:number,pageSize:number) => {
  return axiosInstance.get(`/admin/subscription/get-subscription?pageNumber=${pageNo}&pageSize=${pageSize}`);
};

export const subscribeAPI = (userId: number, id: number) =>
  axiosInstance.post(`/admin/subscriber/subscribe/${userId}`, id);

export const unsubscribeAPI = (userId: number, subscriptionId: number) => 
  axiosInstance.put(`/admin/subscriber/unsubscribe/${userId}`, subscriptionId);

  export const usersubscriptionAPI = (userId: number) => 
  axiosInstance.get(`/admin/subscriber/get-subscriber/${userId}`);