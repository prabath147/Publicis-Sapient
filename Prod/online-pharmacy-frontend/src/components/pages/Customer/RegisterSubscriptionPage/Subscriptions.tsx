import { Badge, Button, Card, Center, Container, Grid, Group, LoadingOverlay, Text, Title } from "@mantine/core";
import { showNotification } from "@mantine/notifications";
import { useEffect, useState } from "react";
import { useAppSelector } from "../../../../app/hooks";
import { getUserData } from "../../Auth/Login/UserSlice";
import { RegisterSubscription } from "./models";
import { unsubscribeAPI, usersubscriptionAPI } from "./SubscriptionAPI";
// import { useNavigate } from "react-router-dom";

export default function Subscriptions() {

  const [subscriptions, setSubscriptions] = useState<RegisterSubscription[]>([]);
  const [loading, setLoading] = useState(true);
  const user = useAppSelector(getUserData);

  useEffect(() => {
    usersubscriptionAPI(user.id).then((response) => {
      // console.log(response.data.userSubsSet);
      let userSubscriptions = [];
      userSubscriptions = response.data.userSubsSet.map((item) => {
        return ((item.subscriptions));
      })
      // console.log(userSubscriptions);
      setSubscriptions(userSubscriptions);
    }).catch((error) => {
      console.log('Error' + error.message);
    });
    setLoading(false);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [])

  const handleClick = (id) => {
    unsubscribeAPI(user.id, id).then((response) => {
      showNotification({ message: "Unsubscribed!", color: "green", autoClose: 500 });
      usersubscriptionAPI(user.id).then((response) => {
        console.log(response.data.userSubsSet);
        let userSubscriptions = [];
        userSubscriptions = response.data.userSubsSet.map((item) => {
          return ((item.subscriptions));
        })
        console.log(userSubscriptions);
        setSubscriptions(userSubscriptions);
      }).catch((error) => {
        console.log('Error' + error.message);
      });
    }).catch((error) => {
      console.log('Error' + error.message);
      showNotification({ message: "Something went wrong, please try again!", color: "red" });
    });
  }

  return <> {loading ? (<LoadingOverlay
    loaderProps={{ size: "md", color: "blue", variant: "oval" }}
    overlayOpacity={0.3}
    overlayColor="#c5c5c5"
    visible={loading}
  />) : (<Container>
    <Title align="center" order={2}>
      Your Subscriptions
    </Title><br />
    <Center>
      <Grid columns={90} sx={{ width: '100%' }}>

        {subscriptions.map((item) => {
          return (
            <Grid.Col sm={60} md={30}>
              <Card shadow="sm" p="md" style={{ width: '100%', height: 300, marginTop: '10px', marginBottom: '10px' }} radius="md" withBorder>
                <Group sx={{ width: '100%' }} position="apart">
                  <Text style={{ fontSize: 22, fontWeight: 700, marginTop: 10 }}>{item.subscriptionName}<br />{"â‚¹" + item.subscriptionCost}</Text>
                  <Text size="md" weight={700} style={{ marginBottom: 10 }}>
                    <Badge>{item.startDate + "-" + item.endDate}</Badge>
                  </Text>
                </Group>

                <Group position="apart">

                  <Group style={{ marginBottom: 10 }} spacing="xs">
                    {item.benefits.one_day_delivery === true ? (<Badge variant="gradient" gradient={{ from: '#bd0000', to: 'red' }}>1 DAY DELIVERY</Badge>) : ""}
                    {item.benefits.delivery_discount !== 0 ? (<Badge variant="gradient" gradient={{ from: '#bd0000', to: 'red' }}>{item.benefits.delivery_discount + "% OFF ON DELIVERY"}</Badge>) : ""}
                  </Group>

                </Group>
                <Text lineClamp={2}>{item.description}</Text>
                <Center mt="xs">
                  {/* <Button disabled size="xs" color="green">Subscribed</Button> */}
                  <Button onClick={() => handleClick(item.subscriptionId)} style={{ marginLeft: 10 }} size="xs" variant="gradient">Unsubscribe</Button>
                </Center>

              </Card>
            </Grid.Col>
          );
        })}
      </Grid>
    </Center>
    <br />
    {/* <Center>
      <Button size="sm" onClick={RegisterSubscriptionPage}>
        Upgrade
      </Button>
    </Center> */}

  </Container>)}
  </>
}