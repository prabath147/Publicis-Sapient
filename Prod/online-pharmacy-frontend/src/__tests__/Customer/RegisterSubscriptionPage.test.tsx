import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import { store } from "../../app/store";
import RegisterSubscriptionPage from "../../components/pages/Customer/RegisterSubscriptionPage/RegisterSubscriptionPage";
import mockAxios from "../../__mocks__/axios";

describe("RegisterSubscriptionPage", () => {
  jest.useFakeTimers().setSystemTime(new Date("2020-01-01"));

  afterEach(() => {
    jest.resetAllMocks();
  });

  it("component created", () => {
    mockAxios.get.mockImplementationOnce(() => Promise.reject());

    const component = renderer.create(
      <Provider store={store}>
        <RegisterSubscriptionPage />
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  // TODO should update it
  // it('elements are in the document', () => {
  //   mockAxios.get.mockImplementationOnce(() =>
  //     Promise.resolve({
  //       data: {

  //       }
  //     }),
  //   )

  //   render(
  //     <Provider store={store}>
  //       <RegisterSubscriptionPage />
  //     </Provider>
  //   )
  //   const title = screen.getByRole('heading', {
  //     name: /available subscriptions/i
  //   })
  //   const submitButton = screen.getByRole('button', {
  //     name: /submit/i
  //   })

  //   expect(title).toBeInTheDocument();
  //   expect(submitButton).toBeInTheDocument();

  // })
});
