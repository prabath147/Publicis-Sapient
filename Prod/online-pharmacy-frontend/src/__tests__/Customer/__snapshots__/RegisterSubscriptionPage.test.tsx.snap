// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterSubscriptionPage component created 1`] = `
Array [
  " ",
  <div
    className="mantine-LoadingOverlay-root mantine-137kca9"
    data-testid="Loading"
    style={
      Object {
        "zIndex": 400,
      }
    }
  >
    <svg
      className="mantine-1avyp1d"
      height="36px"
      role="presentation"
      stroke="#228be6"
      style={
        Object {
          "zIndex": "calc(400 + 1)",
        }
      }
      viewBox="0 0 38 38"
      width="36px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fillRule="evenodd"
      >
        <g
          strokeWidth="5"
          transform="translate(2.5 2.5)"
        >
          <circle
            cx="16"
            cy="16"
            r="16"
            strokeOpacity=".5"
          />
          <path
            d="M32 16c0-9.94-8.06-16-16-16"
          >
            <animateTransform
              attributeName="transform"
              dur="1s"
              from="0 16 16"
              repeatCount="indefinite"
              to="360 16 16"
              type="rotate"
            />
          </path>
        </g>
      </g>
    </svg>
    <div
      className="mantine-Overlay-root mantine-8uy2jv"
    />
  </div>,
  <div
    className="mantine-Group-root mantine-fosyyf"
    data-testid="paginationT"
    role="navigation"
  >
    <button
      aria-disabled={true}
      className="mantine-Group-child mantine-Pagination-item mantine-1bnfkxz"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="none"
        height={16}
        viewBox="0 0 16 16"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.219 8l3.3 3.3-.943.943L5.333 8l4.243-4.243.943.943-3.3 3.3z"
          fill="currentColor"
        />
      </svg>
    </button>
    <button
      aria-current="page"
      aria-label={null}
      className="mantine-Group-child mantine-Pagination-item mantine-1bnfkxz"
      data-active={true}
      onClick={[Function]}
      tabIndex={0}
      type="button"
    >
      1
    </button>
    <button
      aria-disabled={true}
      className="mantine-Group-child mantine-Pagination-item mantine-1bnfkxz"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="none"
        height={16}
        viewBox="0 0 16 16"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.781 8l-3.3-3.3.943-.943L10.667 8l-4.243 4.243-.943-.943 3.3-3.3z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>,
]
`;
